pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -358,20 +391,23 @@ func (c *EtcdCertSignerController) ensureBundles(ctx context.Context, recorder e
 
 	safe, err := c.quorumChecker.IsSafeToUpdateRevision()
 	if err != nil {
-		return nil, nil, fmt.Errorf("EtcdCertSignerController.ensureBundles can't evaluate whether quorum is safe: %w", err)
+		return nil, nil, false, fmt.Errorf("EtcdCertSignerController.ensureBundles can't evaluate whether quorum is safe: %w", err)
 	}
-
 	if !safe {
-		return nil, nil, fmt.Errorf("skipping EtcdCertSignerController.ensureBundles reconciliation due to insufficient quorum")
+		return nil, nil, false, fmt.Errorf("skipping EtcdCertSignerController.ensureBundles reconciliation due to insufficient quorum")
 	}
-	_, changed, err := resourceapply.ApplyConfigMap(ctx, c.configMapClient, recorder, configMap)
+
+	_, rolloutTriggered, err = resourceapply.ApplyConfigMap(ctx, c.configMapClient, recorder, configMap)
 	if err != nil {
-		return nil, nil, fmt.Errorf("could not apply bundle configmap: %w", err)
+		return nil, nil, rolloutTriggered, fmt.Errorf("could not apply bundle configmap: %w", err)
 	}
 
-	if changed {
-		// TODO(thomas): set the rollout bundle revision in the CEO status
-		// TODO(thomas): we need to notify the outside caller that it's not safe to rotate the leafs just yet
+	if rolloutTriggered {
+		// note that even when this operation fails, the triggered revision will not generate leaf certificates
+		// this is guarded by the next sync loops first check against a rolling out revision in ceohelpers.CurrentRevision
+		if err = c.updateCertRevision(ctx, currentRevision); err != nil {
```

Comment:
The bundle is updated before the operator status and should cause revision 7, yes. So when it fails to persist the next sync invocation will _likely_ go into the static pod rollout detection. Once that revision is rolled out, it _should_ be safe to generate the leaf certs. 

In reality, I'm sure this is not as happy path, we know from the AI failures that sometimes revisions just can stomp right in between. 

I have an alternative solution where we store this as part of the bundle to avoid this non-transactional double object update, but that would also require library-go changes, so I held off on this for now.  The leaf certs also have the signer certificate and resource version in their annotation, so we could also try to match from the side.

Reply:
sorry, we own our bundled bundle configmap now :) so we can scrap that part of ETCD-633 and add that annotation on the apply.

Reply:
checkout https://github.com/openshift/cluster-etcd-operator/pull/1275 - I'll run some tests on the other PR, so we can see whether this works better

Reply:
works just as well, so we can either merge #1275 directly or have the commit as a separated PR, let me know what you prefer :)

Reply:
Thanks for the walkthrough!
And yep just finished looking at https://github.com/openshift/cluster-etcd-operator/pull/1275 and I think that's much cleaner with the revision being stored on the configmap annotation rather than a separate object. So let's merge that in favor of this one.

Reply:
lovely, then let's close :)